src/DataPrinter.java

```diff
@@ -0,0 +1,35 @@
+public class DataPrinter {
+    Node next;
```

Comment:
You are using here probably this class. https://docs.oracle.com/javase/8/docs/api/org/w3c/dom/Node.html

I don't think that it is what you intended.

Please rename the class s/DataPrinter/Node/ so that you will use the Node class you defined